### DISCUSSION TITLE
Supported versions of quarkus

### DISCUSSION THREAD

**Discussion by robp94** [jobrunr/jobrunr#1103] [Q&A] [answered]
We are evaluating JobRunr and use Quarkus for our applications. Could you clarify which versions of Quarkus you plan to support? Are there any differences between the free and paid versions in terms of supported Quarkus versions?

Additionally, the current Quarkus extension of JobRunr is based on the LTS version 3.8.3 from March. The latest LTS version is 3.8.5 from June, with 3.13.3 as the latest overall version, and 3.14.x expected soon. We generally try to adopt the latest Quarkus releases, which can include breaking changes. This makes your support for recent versions important to us.

 https://github.com/quarkusio/quarkus/wiki/Release-Planning
 
 Furthermore, while trying to integrate JobRunr in one of our services we got an exception on startup `java.lang.ClassNotFoundException: io.quarkus.smallrye.health.runtime.SmallRyeHealthRecorder`, adding the smallrye-health extension fixes this. 

But based on the [documentation](https://www.jobrunr.io/en/documentation/configuration/quarkus/), we would have assumed that it would be optional, since only some JSON framework is specified as required. If it is not optional, maybe this can be added to the documentation? Else, here is a small reproducer based on your quarkus example: https://github.com/robp94/example-quarkus/tree/bug/smallrye-health

**Comment by auloin** [comment, marked as answer]
Hi @robp94, we try to support the LTS version of Quarkus but the versions change so quickly that it's hard for us to keep up.

The `SmallRyeHealthRecorder` issue seems to be an issue in Quarkus ([see this Quarkus Zulip chat](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/JobRunr.3A.20upgrading.20custom.20extension.20from.203.2E7.2E3.20to.203.2E8.2E2/near/425940335)). We are unable to work arround it at the moment.

**Reply by robp94** [reply]
Thanks for the fast answer.

**Comment by rdehuyss** [comment]
FYI - we just tested JobRunr against Quarkus 3.14.1 and all works well.
